historical-adherence-query/java/Main.java

```diff
@@ -1,64 +1,55 @@
 import com.google.common.base.Stopwatch;
 import com.mypurecloud.sdk.v2.ApiClient;
 import com.mypurecloud.sdk.v2.ApiException;
+import com.mypurecloud.sdk.v2.ApiResponse;
 import com.mypurecloud.sdk.v2.Configuration;
 import com.mypurecloud.sdk.v2.PureCloudRegionHosts;
 import com.mypurecloud.sdk.v2.api.UsersApi;
 import com.mypurecloud.sdk.v2.api.WorkforceManagementApi;
-import com.mypurecloud.sdk.v2.extensions.notifications.NotificationHandler;
-import com.mypurecloud.sdk.v2.extensions.notifications.NotificationListener;
-import com.mypurecloud.sdk.v2.model.User;
-import com.mypurecloud.sdk.v2.model.UserMe;
-import com.mypurecloud.sdk.v2.model.WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice;
-import com.mypurecloud.sdk.v2.model.WfmHistoricalAdherenceQuery;
-import com.mypurecloud.sdk.v2.model.WfmHistoricalAdherenceResponse;
+import com.mypurecloud.sdk.v2.model.*;
+import com.mypurecloud.sdk.v2.extensions.notifications.*;
 import com.neovisionaries.ws.client.WebSocketException;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Calendar;
-import java.util.Collections;
-import java.util.Date;
-import java.util.HashMap;
-import java.util.List;
-import java.util.NoSuchElementException;
+import java.util.*;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 import java.util.function.Supplier;
 
-public class Main {
+public class HistoricalAdherenceQuery {
 
     public static void main(String[] args) throws IOException, ApiException, InterruptedException, WebSocketException,
-        TimeoutException {
+            TimeoutException {
 
-        // Set client credentials
-        String clientId = System.getenv("GENESYS_CLOUD_CLIENT_ID");
-        String clientSecret = System.getenv("GENESYS_CLOUD_CLIENT_SECRET");
+        // Token should be generated from Authorization with user context (Implicit Grant or Code Authorization).
+        String accessToken = "--- ACCESS TOKEN ---";
```

Comment:
Just out of curiosity.  Why did you switch from a Client Credential grant to one of the other OAuth mechanisms?

Reply:
Hi John, in lines 167-170, the code creates a Notification Listener using a user-context dependent topic.

```
public HistoricalAdherenceEventListener(String userId) {
            // The topic to which the notification handler subscribes
            this.topic = String.format("v2.users.%s.workforcemanagement.historicaladherencequery", userId);
        }
```

Also in line 34-37, there is code to get the current user whose id will be used for the topic above.
```
// Create an instance of UsersApi to retrieve your own userId
        UsersApi usersApi = new UsersApi();
        UserMe me = usersApi.getUsersMe(Collections.emptyList(), null);
        String myId = me.getId();
```

My guess is that the Authorization part of the code was included only afterwards because the original code wouldn't run due to Client Credentials token not being able to call the UserMe and subscribe to the topic. 

I'm not a 100% sure if there's an implementation that doesn't require user context but for the meantime to get it 'working' I just replaced the authorization with token. One of the working tutorials also does the same thing (presence - Java) so I'm not sure what our standard should be for such tutorials. Adding webpage capability (for implicit or code auth) on top of the tutorial content itself seems 'overkill', but not sure what would be the proper way to go about it.

Reply:
Cool.  I agree.  Tutorials should be simple and to the point.  Leave the more complicated discussions for user guides.